### PR TITLE
Allow exporting const variables when `strictNullChecks` is on

### LINKED ISSUE #15329: Exporting `const` variables still raises an error if `strictNullChecks` is on
**TypeScript Version:** nightly (2.3.0-dev.20170423)

This issue is a follow up on https://github.com/Microsoft/TypeScript/issues/14679. In the current nightly, exporting `const` variables is allowed unless `strictNullChecks` is enabled.

**Code**

```ts
// @strictNullChecks: true
export { x }
//       ~
// error TS2454: Variable 'x' is used before being assigned.
const x = 'x'
```

**Expected behavior:**
Exporting const variables should be allowed regardless of `strictNullChecks`.

### PR DESCRIPTION
**Pull request by gcnew**
Fixes #15329

### CODE REVIEW

File: tests/baselines/reference/exportBinding.errors.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+tests/cases/conformance/es6/modules/exportConsts.ts(3,16): error TS2448: Block-scoped variable 'x' used before its declaration.
+tests/cases/conformance/es6/modules/exportConsts.ts(3,16): error TS2454: Variable 'x' is used before being assigned.
+tests/cases/conformance/es6/modules/exportVars.ts(3,16): error TS2454: Variable 'y' is used before being assigned.
+
+
+==== tests/cases/conformance/es6/modules/exportConsts.ts (2 errors) ====
+    export { x }
+    export { x as xx }
+    export default x;
```
Comment (gcnew) [inline tests/baselines/reference/exportBinding.errors.txt]: `export default` should still be an error, right?

Comment (DanielRosenwasser) [inline tests/baselines/reference/exportBinding.errors.txt]: According to what I'm seeing in the spec, I believe that is correct. The declaration evaluates and may have a side-effect, whereas the others are simple bindings.

**Comment by DanielRosenwasser** [comment]
👍

File: tests/baselines/reference/exportBinding.errors.txt (hunk 1 above)
Comment (gcnew) [inline tests/baselines/reference/exportBinding.errors.txt]: Yes, that's what I understood as well. Thank you!

**Pull request merged**